xrpl/asyncio/account/main.py

```diff
@@ -97,11 +99,30 @@ async def get_account_root(
 
     Returns:
         The AccountRoot dictionary for the address.
+
+    Raises:
+        XRPLRequestFailureException: if the rippled API call fails.
     """
-    account_info = await get_account_info(address, client, ledger_index)
+    if is_valid_xaddress(address):
+        address, _, _ = xaddress_to_classic_address(address)
```

Comment:
It's bad practice to overwrite a parameter because it can be very surprising to have the parameter change (Especially if it's an object, in which case the change could cause side effects)

Instead I'd recommend creating a variable inside the function and setting/changing it instead of the parameter.